cmd/query/app/flags.go

```diff
@@ -150,13 +154,13 @@ func (qOpts *QueryOptions) BuildQueryServiceOptions(storageFactory storage.BaseF
 	return opts
 }
 
-func (qOpts *QueryOptions) BuildQueryServiceOptionsV2(storageFactory storage.BaseFactory, logger *zap.Logger) *v2querysvc.QueryServiceOptions {
+func (qOpts *QueryOptions) BuildQueryServiceOptionsV2(storageFactory *storage.Factory, logger *zap.Logger) *v2querysvc.QueryServiceOptions {
 	opts := &v2querysvc.QueryServiceOptions{}
 
-	ar, aw := v1adapter.InitializeArchiveStorage(storageFactory, logger)
+	ar, aw := storageFactory.InitArchiveStorage(logger)
 	if ar != nil && aw != nil {
-		opts.ArchiveTraceReader = ar
-		opts.ArchiveTraceWriter = aw
+		opts.ArchiveTraceReader = v1adapter.NewTraceReader(ar)
+		opts.ArchiveTraceWriter = v1adapter.NewTraceWriter(aw)
 	} else {
```

Comment:
@yurishkuro Any thoughts on how we should test this block? It will go through the real factory and the archive storage is only enabled for cassandra/es, which require running instances to initialize the session.

Reply:
same as with grpc handler - accept interface instead of struct (define interface locally).

Reply:
if you end up reusing that interface in many places, then perhaps define it next to meta-factory, e.g. `type MetaFactory interface`